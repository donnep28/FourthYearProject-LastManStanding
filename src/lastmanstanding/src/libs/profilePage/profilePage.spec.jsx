import React from 'react';
import {shallow} from 'enzyme';
import ProfilePage from './profilePage';
import axios from 'axios';

describe('profilePage', () => {
  beforeAll(() => {
    jest.resetAllMocks();
  });

  const props = {
    user: {
      username: 'philip',
      attributes: {
        'sub': "ecbc56df-d81a-410d-b399-4421b55ced32",
        'email': "philip.donnelly28@mail.dcu.ie",
        'email_verified': true,
      },
    }
  };

  describe('Snapshot', () => {
    it('renders correctly', () => {
      const component = shallow(
          <ProfilePage {...props}/>,
      );
      expect(component).toMatchSnapshot();
    });
  });
});