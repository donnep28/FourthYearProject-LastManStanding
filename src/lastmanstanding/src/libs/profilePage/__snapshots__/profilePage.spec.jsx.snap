// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profilePage Snapshot renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="row"
  spacing={2}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={3}
    xs={12}
  >
    <WithStyles(ForwardRef(Card))>
      <WithStyles(ForwardRef(CardHeader))
        avatar={
          <ForwardRef(WithStyles)
            aria-label="account"
            className="makeStyles-avatar-2"
          >
            <UNDEFINED />
          </ForwardRef(WithStyles)>
        }
        title="Account"
      />
      <WithStyles(ForwardRef(Divider)) />
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(List))>
          <WithStyles(ForwardRef(CircularProgress))
            style={
              Object {
                "color": "#490050",
              }
            }
          />
        </WithStyles(ForwardRef(List))>
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(Card))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={3}
    xs={12}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={12}
      xs={12}
    >
      <WithStyles(ForwardRef(Card))>
        <WithStyles(ForwardRef(CardHeader))
          avatar={
            <ForwardRef(WithStyles)
              aria-label="results"
              className="makeStyles-avatar-2"
            >
              <UNDEFINED />
            </ForwardRef(WithStyles)>
          }
          title="Results"
        />
        <WithStyles(ForwardRef(Divider)) />
        <WithStyles(ForwardRef(CardContent))>
          <WithStyles(ForwardRef(List))>
            <WithStyles(ForwardRef(CircularProgress))
              style={
                Object {
                  "color": "#490050",
                }
              }
            />
          </WithStyles(ForwardRef(List))>
        </WithStyles(ForwardRef(CardContent))>
      </WithStyles(ForwardRef(Card))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={12}
      xs={12}
    >
      <WithStyles(ForwardRef(Card))>
        <WithStyles(ForwardRef(CardHeader))
          avatar={
            <ForwardRef(WithStyles)
              aria-label="active competitions"
              className="makeStyles-avatar-2"
            >
              <UNDEFINED />
            </ForwardRef(WithStyles)>
          }
          title="Active Competitions"
        />
        <WithStyles(ForwardRef(Divider)) />
        <WithStyles(ForwardRef(CardContent))>
          <WithStyles(ForwardRef(List))>
            <WithStyles(ForwardRef(CircularProgress))
              style={
                Object {
                  "color": "#490050",
                }
              }
            />
          </WithStyles(ForwardRef(List))>
        </WithStyles(ForwardRef(CardContent))>
      </WithStyles(ForwardRef(Card))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={6}
    xs={12}
  >
    <WithStyles(ForwardRef(Card))
      style={
        Object {
          "height": "650px",
          "overflow": "scroll",
        }
      }
    >
      <WithStyles(ForwardRef(CardHeader))
        avatar={
          <ForwardRef(WithStyles)
            aria-label="team news"
            className="makeStyles-avatar-2"
          >
            <UNDEFINED />
          </ForwardRef(WithStyles)>
        }
        title="Your Favourite Teams News!"
      />
      <WithStyles(ForwardRef(Divider)) />
      <WithStyles(ForwardRef(CardContent))>
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": "60vh",
              "justifyContent": "center",
            }
          }
        >
          <WithStyles(ForwardRef(CircularProgress))
            style={
              Object {
                "color": "#490050",
                "marginRight": "12px",
              }
            }
          />
        </div>
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(Card))>
    <SelectFavouriteTeam
      myTeam={false}
      selectTeam={[Function]}
      setmyTeam={[Function]}
      teams={
        Object {
          "Arsenal FC": "arsenal",
          "Aston Villa FC": "astonvilla",
          "Brighton & Hove Albion FC": "brightonandhovealbion",
          "Burnley FC": "burnley",
          "Chelsea FC": "chelsea",
          "Crystal Palace FC": "crystalpalace",
          "Everton FC": "everton",
          "Fulham FC": "fulham",
          "Leeds United FC": "leedsunited",
          "Leicester City FC": "leicestercity",
          "Liverpool FC": "liverpool",
          "Manchester City FC": "manchestercity",
          "Manchester United FC": "manchesterunited",
          "Newcastle United FC": "newcastleunited",
          "Sheffield United FC": "sheffieldunited",
          "Southampton FC": "southampton",
          "Totenham Hotspur FC": "totenhamhotspur",
          "West Bromwich Albion FC": "westbromwichalbion",
          "West Ham United FC": "westhamunited",
          "Wolverhampton Wanderers FC": "wolverhamptonwanderers",
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
